Source/RapyutaSimulationPlugins/Private/Sensors/ROS2CameraComponent.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright 2020-2021 Rapyuta Robotics Co., Ltd.
+
+#include "Sensors/ROS2CameraComponent.h"
+
+UROS2CameraComponent::UROS2CameraComponent()
+{
+    // component initialization
+    SceneCaptureComponent = CreateDefaultSubobject<USceneCaptureComponent2D>(TEXT("SceneCaptureComponent"));
+    RenderTarget = CreateDefaultSubobject<UTextureRenderTarget2D>(TEXT("RenderTarget"));
+
+    // Initialize capture and texture component
+    SceneCaptureComponent->SetupAttachment(this);
+}
+
+void UROS2CameraComponent::BeginPlay()
+{
+	Super::BeginPlay();	
+    Init();
+}
+
+void UROS2CameraComponent::Init()
+{
+    SceneCaptureComponent->FOVAngle = FieldOfView;
+    SceneCaptureComponent->OrthoWidth = OrthoWidth;
+    
+    RenderTarget->InitCustomFormat(Width, Height, EPixelFormat::PF_B8G8R8A8, true);
+    SceneCaptureComponent->TextureTarget = RenderTarget;
+
+    // Initialize image data
+    Data.width = Width;
+    Data.height = Height;
+    Data.encoding = Encoding;
+    Data.step = Width * 3; // todo should be variable based on encoding
+    Data.data.AddUninitialized(Width * Height * 3);
+
+    // Node and publisher initialize
+    FActorSpawnParameters SpawnParamsNode;
+    Node = GetWorld()->SpawnActor<AROS2Node>(AROS2Node::StaticClass(), SpawnParamsNode);

```

Comment:
If we are using a default `FActorSpawnParameters`, maybe no need to pass it in, as well as the `AROS2Node::StaticClass()`, which is already deduced from the template type as `AROS2Node`:
```suggestion
   Node = GetWorld()->SpawnActor<AROS2Node>();
```